supervision/detection/core.py

```diff
@@ -133,6 +137,7 @@ class simplifies data manipulation and filtering, providing a uniform API for
     class_id: Optional[np.ndarray] = None
     tracker_id: Optional[np.ndarray] = None
     data: Dict[str, Union[np.ndarray, List]] = field(default_factory=dict)
+    metadata: Dict[str, Any] = field(default_factory=dict)
 
     def __post_init__(self):
         validate_detections_fields(
```

Comment:
In `Detections.empty()` you set the metadata to `metadata=metadata if metadata is not None else`. Let's add some validation, making sure it's a `dict`.